x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/types.ts

```diff
@@ -17,11 +17,13 @@ export const DEFAULT_META = {
   total: 0,
 };
 
-export const convertMetaToPagination = (meta: Meta) => ({
-  pageIndex: meta.from - 1,
-  pageSize: meta.size,
-  totalItemCount: meta.total,
-});
+export const convertMetaToPagination = (meta: Meta) => {
```

Comment:
question: We now have a third different pagination standard in our ecosystem (old Enterprise Search, Kibana Pagination type, and this one). That seems subtoptimal. Is there something we can do to standardize on something again?

Reply:
cc: @demjened @kderusso

Reply:
If this endpoint is built in Enterprise Search, it should adhere to the [Enterprise Search pagination standards](https://github.com/elastic/ent-search/blob/685d3f1a7243b8ee250e8a46668b8af53818b385/lib/pagination.rb) (or folks should have a discussion about changing those). For Kibana endpoints we have a [pagination standard](https://github.com/elastic/ent-search/blob/685d3f1a7243b8ee250e8a46668b8af53818b385/lib/pagination.rb) that makes interacting with EUI tables a lot easier.

Reply:
it’s in enterprise-search, but will be using grape soon.

Reply:
Thanks for the link on pagination standards. I think it's worth a discussion here, because we were trying to adhere to API conventions used in other parts of the platform as well. If we want to change it we can create a ticket for it.

Reply:
I think it's worth discussing what we want to do in terms of standards here. For context: the current Enterprise Search standard is essentially a legacy standard from back when we were still Swiftype and has some weird quirks. Our Kibana standard is really aimed at making it as easy as possible to work with Elastic UI tables for pagination.

It's worth taking into account what the main point of these APIs is as well. If we're expecting them to be used by users directly in the same way they might use Elasticsearch APIs, then adhering to Elasticsearch's query standards probably makes a lot of sense. But if the primary consumer is going to be Kibana (or some other UI we build), it probably makes more sense to standardize on one of the existing pagination standards.

Reply:
Raised the subject in Slack: https://elastic.slack.com/archives/C02U50QNEAG/p1673966205671379